File: starlore-back/src/main/java/com/robin/blogback/entity/Resume.java
package com.robin.blogback.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("resumes")
public class Resume {
    @TableId(type = IdType.AUTO)
    private Integer id;

    @TableField("user_id")
    private Integer userId;

    private String title;
    private String template;

    private String name;

    @TableField("job_title")
    private String jobTitle;

    private String phone;
    private String email;

    @TableField("photo_url")
    private String photoUrl;

    private String content;
    private String status;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;
}

File: starlore-py/app/schemas/resume.py
"""简历相关 Schema。"""

from datetime import datetime

from pydantic import BaseModel


class ResumeItem(BaseModel):
    id: int
    userId: int
    title: str | None = None
    template: str | None = None
    name: str | None = None
    jobTitle: str | None = None
    phone: str | None = None
    email: str | None = None
    photoUrl: str | None = None
    content: str | None = None
    status: str | None = None
    createdAt: datetime | None = None
    updatedAt: datetime | None = None


class ResumeResponse(BaseModel):
    data: ResumeItem
    message: str


class ResumeListResponse(BaseModel):
    data: list[ResumeItem]


class ResumeDetailResponse(BaseModel):
    data: ResumeItem


